### PR TITLE
Initial implementation of addon store, api, and collection rendering …

### PR DESCRIPTION
**Pull request by RonSpawnson**
Getting my feet wet with by extending the project to add basic support for fetching and showing addons (addon store, api, and collection rendering). 

You can test it out by viewing the "recent" addons page. 

Caveat: The render is rough and needs styling/designing. I'm printing the values out right now as a stand-in placeholder.

### CODE REVIEW

File: client/components/addons/addon-collection/addon-collection.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+const AddonCollection = ({ addons }) => {
+	const renderedAddonElements = addons.map(addon => (
+		<div id={addon.name} key={addon.name}>
```
Comment (samhh) [inline client/components/addons/addon-collection/addon-collection.js]: What's the `id` for?

Comment (samhh) [reply client/components/addons/addon-collection/addon-collection.js]: Also semantically this should probably be a set of `<li>` wrapped in a `<ul>`.

File: client/components/addons/addon-collection/addon-collection.sass [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.list
+	padding-left: 20px
+
+	li
+		padding: 5px 0
```
Comment (samhh) [inline client/components/addons/addon-collection/addon-collection.sass]: Don't think you're using any of these styles at the moment?

File: client/components/addons/addon-collection/container.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	static propTypes = {
+		addon: PropTypes.object.isRequired
+	}
+
+	render () {
+		const { addon } = this.props
+
+		return (
+			<AddonCollectionPage
+				addons={addon.addonsSortedByMostRecent}
```
Comment (samhh) [inline client/components/addons/addon-collection/container.js]: You possibly shouldn't be specifying the addons here as now if you want to reuse this component on another page you're tied to having them sorted by most recent. Perhaps pass them in as a prop, and let the pages handle that contextually?

File: client/pages/addons/recent/recent.js [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
 
+import AddonCollection from '../../../components/addons/addon-collection'
 import AddonsNav from '../addons-nav'
 
-const RecentAddonsPage = () => (
-	<div>
-		<AddonsNav />
+class RecentAddonsPage extends Component {
+	static propTypes = {
+		addons: MobxPropTypes.observableArrayOf(MobxPropTypes.observableObject).isRequired
```
Comment (samhh) [inline client/pages/addons/recent/recent.js]: Don't think this is being used? But it probably should be, and passed into `<AddonCollection>` as a prop as I said above.

File: client/pages/addons/recent/recent.sass [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+	border: 2px solid $color-primary
+	border-radius: 1px
+	background: rgba(0, 0, 0, .3)
+
+	> header h1
+		margin-top: 0
+
+	label
+		display: inline-block
+		margin: 0 ($baseline-spacing / 3) ($baseline-spacing / 3) 0
```
Comment (samhh) [inline client/pages/addons/recent/recent.sass]: Don't think you're using any of these styles at the moment?

File: client/stores/addon.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	@action getAddons = async () => {
+		const addons = await axios.get('/api/addons').then(res => res.data.addons)
+
+		runInAction(() => {
+			this.addons = addons
+		})
+	}
+}
+
+export default new AddonStore()
```
Comment (samhh) [inline client/stores/addon.js]: Super duper minor nitpick: Would prefer if it was called `addons` store (plural). User store is singular as it was only supposed to hold the client alone (may also change, it's not scoped out).

**Review by samhh — CHANGES_REQUESTED** [review]
See notes. It's 95% there 👍

File: client/components/addons/addon-collection/addon-collection.js (hunk 1 above)
Comment (RonSpawnson) [reply client/components/addons/addon-collection/addon-collection.js]: id is for nothing right now - I'll remove it as it has no use. Will also update to unordered list

File: client/components/addons/addon-collection/addon-collection.sass (hunk 2 above)
Comment (RonSpawnson) [reply client/components/addons/addon-collection/addon-collection.sass]: correct, I'll remove for clarity

File: client/components/addons/addon-collection/container.js (hunk 3 above)
Comment (RonSpawnson) [reply client/components/addons/addon-collection/container.js]: Yep, good catch. Will be needed to be able to reuse component for other orderings

File: client/pages/addons/recent/recent.js (hunk 4 above)
Comment (RonSpawnson) [reply client/pages/addons/recent/recent.js]: Yep, I agree

File: client/stores/addon.js (hunk 6 above)
Comment (RonSpawnson) [reply client/stores/addon.js]: Yeah I went back and forth on this. I've seen collections referred to with either naming convention, so I just picked one. Looking at it now I agree with your reasoning for the plural and will move to that

**Comment by RonSpawnson** [comment]
I appreciate the feedback / nitipicks, especially as I get acquainted with the technology/project structure. 

I'll work on addressing the above and issuing a new diff once ready.

**Comment by RonSpawnson** [comment]
Ready for re-review / merge.

Long term strategy may be to migrate to filterable/sortable table. Two pages - Store and Installed, or something like that

**Review by samhh — APPROVED** [review]

**Pull request merged**